### PR TITLE
F/eslint from tempates js

### PR DESCRIPTION
**Pull request by dbouwman**
- Remove tslint.json
- add eslint & plugins
- ignore files in `packages/**/{test,dist}`
- disable the following rules
```
"rules": {
      "@typescript-eslint/no-explicit-any": "off",
      "@typescript-eslint/no-unused-vars": "off",
      "@typescript-eslint/ban-types": "off",
      "@typescript-eslint/explicit-module-boundary-types": "off"
    }
```
Fix others.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,7 +104,7 @@
     "docs:dev:sass": "node-sass --watch --recursive --output docs/build/arcgis-rest-js/css --source-map true --source-map-contents docs/src/sass",
     "docs:dev:js": "cpx \"docs/src/**/{api-search,nav-toggle}.js\" docs/build/arcgis-rest-js -w",
     "docs:srihash": "node docs/generate-srihashes.js",
-    "lint": "tslint --project tsconfig.json",
+    "lint": "eslint ./packages --ext .ts",
     "lint:fix": "tslint --project tsconfig.json --fix",
```
Comment (MikeTschudi) [inline package.json:108]: Should be eslint for lint:fix

File: packages/arcgis-rest-auth/src/federation-utils.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ const arcgisOnlinePortalRegex = /^https?:\/\/(dev|devext|qa|qaext|www)\.arcgis\.
 /**
  * Used to test if a URL is an ArcGIS Online Organization Portal
  */
+// eslint-disable-next-line no-useless-escape
 const arcgisOnlineOrgPortalRegex = /^https?:\/\/(?:[a-z0-9-]+\.maps(dev|devext|qa|qaext)?)?.\arcgis\.com\/sharing\/rest/;
```
Comment (MikeTschudi) [inline packages/arcgis-rest-auth/src/federation-utils.ts:16]: "\a" is a useless escape; we don't need the disable

File: packages/arcgis-rest-routing/src/helpers.ts [hunk 3]
```diff
@@ -70,14 +70,13 @@ export function decompressGeometry(str: string) {
   const points = [];
   let x;
   let y;
-  let strings;
-  let coefficient;
 
   // Split the string into an array on the + and - characters
-  strings = str.match(/((\+|\-)[^\+\-]+)/g);
+  // eslint-disable-next-line no-useless-escape
+  const strings = str.match(/((\+|\-)[^\+\-]+)/g);
```
Comment (MikeTschudi) [inline packages/arcgis-rest-routing/src/helpers.ts:76]: "\+" and "\-" are a useless escapes; we don't need the disable

File: package.json (hunk 1 above)
Comment (dbouwman) [reply package.json]: I removed `lint:fix` for now - guard against a PR w/ all files changed

File: .prettierignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 package.json
-package-lock.json
\ No newline at end of file
+package-lock.json
+SearchQueryBuilder.ts
```
Comment (dbouwman) [inline .prettierignore:3]: Really unclear why prettier just loses it's mind w/ this file and removes _all_ comments. Till we work that out, prettier will ignore it

**Review by MikeTschudi — APPROVED** [review]
Very tidy approach! Thanks for pursuing it.

**Pull request merged**